### ISSUE TITLE
AEM-4 and AEM-5 Treat Injectors Passed to Lambda Expressions as Unclosed

### ISSUE THREAD

**Issue by toniedzwiedz** [Cognifide/AEM-Rules-for-SonarQube#142]
Code where a false positive will be detected:
```
try (InjectorWithContext injector = InjectorUtil.getInjector(ProjectConstants.Aem.INJECTOR_NAME, request)) {
  Optional<MyClass> myInjectedObject = Optional.ofNullable(injector).map(i -> i.getInstance(MyClass.class));
  if (myInjectedObject .isPresent()) {
    // do sth
  }
}
```

SonarQube reports violations of AEM-4 and AEM-5 in the line containing `i -> i.getInstance(MyClass.class)`

Note that the injector is closed automatically thanks to a `try-with-resources` block and that it never leaves the scope of this block.

**Comment by chutch** [comment]
Closing issue as Slice support has been depracated.
